assets/js/RepLog/RepLogApp.js

```diff
@@ -0,0 +1,140 @@
+import React, { Component } from 'react';
+import RepLogs from './RepLogs';
+import PropTypes from 'prop-types';
+import uuid from 'uuid/v4';
+import { getRepLogs, deleteRepLog, createRepLog } from '../api/rep_log_api';
+
+export default class RepLogApp extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = Object.assign({
+            highlightedRowId: null,
+            repLogs: [],
+            numberOfHearts: 1,
+            isLoaded: false,
+            isSavingNewRepLog: false,
+            successMessage: '',
+            newRepLogValidationErrorMessage: '',
+            itemOptions: []
+        }, this.props.initialState);
+
+        this.handleRowClick = this.handleRowClick.bind(this);
+        this.handleAddRepLog = this.handleAddRepLog.bind(this);
+        this.handleHeartChange = this.handleHeartChange.bind(this);
+        this.handleDeleteRepLog = this.handleDeleteRepLog.bind(this);
+    }
+
+    componentDidMount() {
+        getRepLogs()
+            .then((data) => {
+                this.setState({
+                    repLogs: data,
+                    isLoaded: true
+                })
+            });
+    }
+
+    componentWillUnmount() {
+        if (this.clearSuccessMessageTimeout) {
```

Comment:
This, when set as a default to `0` in the constructor doesn't need the if check.